### ISSUE TITLE
Allow internal CV learners to train on tiny (< 10 samples) datasets

### ISSUE THREAD

**Issue by andrewdalpino** [RubixML/ML#95]
Right now, learners that use an internal validation set for things like progress monitoring, early stopping, and neural network snapshotting need at least 1 sample to generate a validation score. With tiny training sets (< 10 samples), there could be a condition that there is not enough data to create a validation set with at least 1 sample given the user's specified hold out ratio. This task is to allow the learner to forego things like progress monitoring and early stopping (in otherwords not create a validation set) if there is not enough data in the training set to generate a validation set.

This will effect MLP Classifier, MLP Regressor and Gradient Boost.
